### PR TITLE
[X86][AVX] Add missing X86ISD::VBROADCAST(v2i64 -> v4i64) isel pattern for AVX1 targets

### PR DESCRIPTION
**Pull request by Maetveis**
An internal bug revealed that this form can be formed from existing combines, and then fail to select. Use the same pattern as v2f64 -> v4f64 (the instructions are moving bits around the actual type does not matter here).

### CODE REVIEW

File: llvm/lib/Target/X86/X86InstrSSE.td [hunk 1]
```diff
@@ -7797,6 +7797,11 @@ let Predicates = [HasAVX1Only] in {
             (VPSHUFDri (VMOV64toPQIrr GR64:$src), 0x44)>;
   def : Pat<(v2i64 (X86VBroadcastld64 addr:$src)),
             (VMOVDDUPrm addr:$src)>;
+
+  def : Pat<(v4i64 (X86VBroadcast v2i64:$src)),
+            (VINSERTF128rr (INSERT_SUBREG (v4i64 (IMPLICIT_DEF)),
+              (v2i64 (VMOVDDUPrr VR128:$src)), sub_xmm),
+              (v2i64 (VMOVDDUPrr VR128:$src)), 1)>;
```
Comment (RKSimon) [inline llvm/lib/Target/X86/X86InstrSSE.td:7804]: Use VPSHUFDri 0x44 instead to reduce domain changes?

**Review by RKSimon — COMMENTED** [review]
Does the v4i32 -> v8i32 equivalent broadcast pattern occur as well?

**Comment by Maetveis** [comment]
> Does the v4i32 -> v8i32 equivalent broadcast pattern occur as well?

I couldn't produce it by changing the types in the IR: https://godbolt.org/z/6nTPjhsqf

File: llvm/lib/Target/X86/X86InstrSSE.td (hunk 1 above)
Comment (Maetveis) [reply llvm/lib/Target/X86/X86InstrSSE.td:7804]: Sure, I will change it.

I'm not versed in these instruction level considerations yet, are you referring to less mixing of "floating point" and integer instructions? Is this sentence in the Intel Optimization Manual relevant?

> When writing SIMD code that works for both integer and floating-point data, use the subset of SIMD convert
instructions or load/store instructions to ensure that the input operands in XMM registers contain data types that
are properly defined to match the instruction.
> Code sequences containing cross-typed usage produce the same result across different implementa-
tions but incur a significant performance penalty. Using SSE/SSE2/SSE3/SSSE3/SSE44.1 instruc-
tions to operate on type-mismatched SIMD data in the XMM register is strongly discouraged.

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**